--- /hub/share/sbtools/apps/emacs-add-ons/src/sb-tools/p4e-cmd.el	2014-10-01 16:02:40.000000000 -0400
+++ /home/jyates/emacs/patched/p4e-cmd.el	2015-08-03 14:36:35.616572000 -0400
@@ -1,6 +1,6 @@
 ;;; p4e-cmd.el --- Command definitions for perforce.
 
-;; Copyright (c) 2005 The MathWorks Inc.
+;; Copyright (c) 2005, 2015 The MathWorks Inc.
 ;;
 ;; < Note: Must provide full copyright info here >
 ;;
@@ -14,53 +14,53 @@
 
 ;;; Info type commands
 ;;
-(p4e-command "info" :command "info" :binding "i")
-(p4e-command "opened" :command "opened" :binding "o" :mode #'p4e-list-mode)
-(p4e-command "files" :command "files" :commandargs '("*") :binding "f" :mode #'p4e-list-mode)
+(p4e-command :name "info" :command "info" :binding "i")
+(p4e-command :name  "opened" :command "opened" :binding "o" :mode #'p4e-list-mode)
+(p4e-command :name  "files" :command "files" :commandargs '("*") :binding "f" :mode #'p4e-list-mode)
 
 ;; p4 changes never returns
 ;;(p4e-command-on-file "changes" :command "changes")
 
 ;; log probably not needed because vc-p4 gives ^X v l which is better because
 ;; you can select a version with f
-(p4e-command-on-file "log" :command "filelog" :commandargs (list "-l") :binding "l")
-(p4e-command "my-pending-changes" :command "changes"
+(p4e-command-on-file :name  "log" :command "filelog" :commandargs (list "-l") :binding "l")
+(p4e-command :name  "my-pending-changes" :command "changes"
 	     :commandargs (list "-s" "pending" "-u" user-login-name))
 
-(p4e-command-form "client" :command "client" :binding "C")
+(p4e-command-form :name "client" :command "client" :binding "C")
 
 ;;; diff'ing
-(p4e-command-on-file "diff" :command "diff" :binding "=" :mode #'diff-mode)
-(p4e-command "have" :command "have" :binding "h" :mode #'p4e-list-mode)
-(p4e-command-on-file-revision "getHaveVersion" :command "print" :commandargs '("-q")
+(p4e-command-on-file :name  "diff" :command "diff" :binding "=" :mode #'diff-mode)
+(p4e-command  :name "have" :command "have" :binding "h" :mode #'p4e-list-mode)
+(p4e-command-on-file-revision  :name "getHaveVersion" :command "print" :commandargs '("-q")
 			      :mode #'p4e-mode-of-origin-RO)
-(p4e-command-ediff-file-revision "ediff" :command "getHaveVersion" :binding "*")
+(p4e-command-ediff-file-revision :name "ediff" :command "getHaveVersion" :binding "*")
 
 ;;; Edit/change type commands
 ;;
-(p4e-command-on-file "edit" :command "edit" :binding "e" :buffer-action 'revert)
-(p4e-command-on-file "revert" :command "revert" :binding "U" :buffer-action 'revert :ask-first-p t)
-(p4e-command-on-file "add" :command "add" :binding "a")
-(p4e-command-on-file "delete" :command "delete" :buffer-action 'kill :ask-first-p t)
+(p4e-command-on-file :name "edit" :command "edit" :binding "e" :buffer-action 'revert)
+(p4e-command-on-file :name  "revert" :command "revert" :binding "U" :buffer-action 'revert :ask-first-p t)
+(p4e-command-on-file :name  "add" :command "add" :binding "a")
+(p4e-command-on-file :name  "delete" :command "delete" :buffer-action 'kill :ask-first-p t)
 
 ;; p4 sync not good with sbsync, etc.
 ;;(p4e-command "sync" :command "sync" :binding "y")
 
 ;;; Submit Stuff
 ;;
-(p4e-command-form "change" :command "change" :binding "c")
-(p4e-command-form "submit" :command "change" :outcommand "submit" :binding "v")
-(p4e-command-form "submit+" :command "change" :outcommand "submit"
+(p4e-command-form :name  "change" :command "change" :binding "c")
+(p4e-command-form :name  "submit" :command "change" :outcommand "submit" :binding "v")
+(p4e-command-form :name  "submit+" :command "change" :outcommand "submit"
 		  :interactivearg '(changeset)
 		  :interactivespec "sChange Set: ")
 
 ;; Help/info
-(p4e-command "clients" :command "clients" :commandargs '("-u") :binding "\C-c"
+(p4e-command :name "clients" :command "clients" :commandargs '("-u") :binding "\C-c"
 	     :interactivearg '(user)
 	     :interactivespec "sClients for user: ")
-(p4e-command "my-changes" :command "changes"
+(p4e-command :name  "my-changes" :command "changes"
 	     :commandargs (list "-u" user-login-name))
-(p4e-command "p4 help COMMAND" :command "help" :binding "H"
+(p4e-command :name "p4 help COMMAND" :command "help" :binding "H"
 	     :interactivearg '(command)
 	     :interactivespec "sHelp for command: ")
 
