--- /hub/share/sbtools/apps/emacs-add-ons/src/sb-tools/p4e.el	2014-10-01 16:02:40.000000000 -0400
+++ /home/jyates/emacs/patched/p4e.el	2015-08-03 14:47:56.982959000 -0400
@@ -1,6 +1,6 @@
 ;;; p4e.el --- Eric's P4 integration mode.
 
-;; Copyright (c) 2005 The MathWorks Inc.
+;; Copyright (c) 2005, 2015 The MathWorks Inc.
 ;;
 ;; Issues:
 ;;   (require 'p4e) causes stata on /mathworks/home/.perforce (from
@@ -52,6 +52,10 @@
 
 (defclass p4e-command (eieio-instance-tracker)
   ((tracking-symbol :initform 'p4e-commands)
+   (name :initarg :name
+         :initform ""
+	 :type string
+	 :documentation "The name of the p4 command to execute")
    (command :initarg :command
 	    :type string
 	    :documentation "The p4 command to execute.")
@@ -83,9 +87,13 @@
   "A Perforce command structure.
 Use the `p4e-exec' method to run this command from a buffer.")
 
+(defmethod p4e-cmd-name ((cmd p4e-command))
+  "Return the user facing name of CMD."
+  (oref cmd name))
+
 (defmethod p4e-fcn-symbol ((cmd p4e-command))
   "Return a symbol that is the Emacs Lisp function for the p4e command CMD."
-  (intern (concat "p4e-" (object-name-string cmd) "-cmd")))  
+  (intern (concat "p4e-" (p4e-cmd-name cmd) "-cmd")))  
 
 (defmethod p4e-bind-command ((cmd p4e-command) &optional special-map)
   "Bind the command CMD into the keymap km."
@@ -102,7 +110,7 @@
 
 (defmethod p4e-easymenu-vector ((cmd p4e-command))
   "Convert CMD into an easymenu vector."
-  (vector (capitalize (object-name-string cmd))
+  (vector (capitalize (p4e-cmd-name cmd))
 	  (p4e-fcn-symbol cmd)
 	  `(p4e-command-valid ,cmd)
 	  )
@@ -150,7 +158,7 @@
   (let ((cmds p4e-commands)
 	(ans nil))
     (while (and cmds (not ans))
-      (if (string= (object-name-string (car cmds)) name)
+      (if (string= (p4e-cmd-name (car cmds)) name)
 	  (setq ans (car cmds)))
       (setq cmds (cdr cmds)))
     ans))
@@ -252,7 +260,7 @@
   (call-next-method)
   ;; Rename the output buffer.
   (let*
-      ((bufName (concat (object-name-string cmd) ": " (oref cmd filesave)))
+      ((bufName (concat (p4e-cmd-name cmd) ": " (oref cmd filesave)))
        (buf (get-buffer bufName)))
     (if buf (kill-buffer buf))
     (rename-buffer bufName)
